module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,40 @@
+#include "compression.hpp"
+
+std::vector<std::pair<uint8_t, uint8_t>> compressOneLine(const std::array<uint8_t, width> &line)
+{
+    std::vector<std::pair<uint8_t, uint8_t>> vect;
+    auto it1 = line.cbegin();
+    auto it2 = line.cend();
```

Comment:
it2 can be inside while loop

Reply:
To be sure, doesn't that mean creating a new object with each 
loop iteration?